pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -856,6 +886,61 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func (r *ReconcileClusterDeployment) selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
+			if cdNamespace, ok := pod.Labels[install.ClusterDeploymentNamespaceLabel]; ok {
+				retval = append(retval, reconcile.Request{NamespacedName: types.NamespacedName{
+					Name:      cdName,
+					Namespace: cdNamespace,
+				}})
+			}
+		}
+	}
```

Comment:
So do we need a predicate to avoid calling this func if the pod is not an install job? If we call the func, and see no labels, or see our labels missing, we just return an empty Request array, that should mean no reconciles happen right? In that case maybe no predicate function is needed.

Reply:
Overall this looks ok lets just remove it from (r *ReconcileClusterDeployment)

Reply:
To the first comment, yes and yes we do not need the predicate function (confirmed with Cesar). For the second comment, Cesar agrees It should not be part of the reconciler object